src/qz/printer/info/CachedPrintServiceLookup.java

```diff
@@ -31,15 +32,32 @@ public static PrintService[] lookupPrintServices() {
         return cachedPrintServices.get();
     }
 
-    private static PrintService innerLookupDefaultPrintService() {
+    private static PrintService wrapDefaultPrintService() {
+        PrintService javaxPrintService = PrintServiceLookup.lookupDefaultPrintService();
+        // If this CachedPrintService already exists, reuse it rather than wrapping a new one
+        CachedPrintService oldCachedPrintService = getMatch(oldPrintServices, javaxPrintService);
+        if (oldCachedPrintService != null) return oldCachedPrintService;
         return new CachedPrintService(PrintServiceLookup.lookupDefaultPrintService());
```

Comment:
This will use a different pointer for the default print service and the copy of it in the listing.

Reply:
hmm, ill test that

Reply:
We could do this 
``` java
private static PrintService wrapDefaultPrintService() {
    PrintService javaxPrintService = PrintServiceLookup.lookupDefaultPrintService();
    // If this CachedPrintService already exists, reuse it rather than wrapping a new one
    CachedPrintService oldCachedPrintService = getMatch(oldPrintServices, javaxPrintService);
    if (oldCachedPrintService != null) return oldCachedPrintService;
    // If its not in our list, force a refresh of the cache
    cachedPrintServices.get(true);
    return getMatch(oldPrintServices, javaxPrintService);
}
```
note, this could in theory return null, though i don't think that scenario is actually possible.

Reply:
This was fixed here https://github.com/qzind/tray/pull/1124/commits/49bb0691c313bc65c7ad056fab3223ba5ea45816#diff-74db1b7f463a02e2ec7078941f6f0f0cf8ef7cb146b0cd95576229069272ec89R42
It also avoids calling `lookupDefaultPrintService` when we are only looking for default.

Reply:
We also need to ensure this works when there is no default print service (e.g. when `PrintServiceLookup.lookupDefaultPrintService()` returns `null`).